pkg/blobserver/s3/enumerate.go

```diff
@@ -48,11 +48,16 @@ func (sto *s3Storage) EnumerateBlobs(ctx context.Context, dest chan<- blob.Sized
 
 	keysGotten := 0
 
-	err := sto.client.ListObjectsV2PagesWithContext(ctx, &s3.ListObjectsV2Input{
+	lo := &s3.ListObjectsV2Input{
 		Bucket:     &sto.bucket,
 		StartAfter: aws.String(sto.dirPrefix + after),
 		MaxKeys:    maxKeys,
-	}, func(page *s3.ListObjectsV2Output, lastPage bool) bool {
+	}
+	if sto.dirPrefix != "" {
```

Comment:
Setting an empty prefix should work just fine? Its the default value for a string anyways. Do we really need to check before?

Reply:
The input is a pointer to a string so I assume that there could be differences in implementation between empty and null string. I didn't want to change what behavior we already had (passing in null string) when there is no prefix.

Reply:
I am not against changing it to always including the pointer.

Reply:
Right, but I am pretty sure an empty prefix works. I'd go with that unless it does not work.

Reply:
done